test/integration/ratio-Encounter-reuseObservationFunction/ratio-Encounter-reuseObservationFunction.test.ts

```diff
@@ -0,0 +1,72 @@
+import { calculate } from '../../../src/calculation/Calculator';
+import { CalculationOptions, PopulationResult } from '../../../src/types/Calculator';
+import { PopulationType } from '../../../src/types/Enums';
+import { getJSONFixture } from '../helpers/testHelpers';
+
+const CALCULATION_OPTIONS: CalculationOptions = {
+  measurementPeriodStart: '2023-01-01',
+  measurementPeriodEnd: '2023-12-31',
+  enableDebugOutput: true
+};
+
+const MEASURE_BUNDLE: fhir4.Bundle = getJSONFixture(
+  'ratio-Encounter-reuseObservationFunction/ratio-Encounter-reuseObservationFunction-bundle.json'
+);
+
+const PATIENT_BUNDLE: fhir4.Bundle = getJSONFixture(
+  'ratio-Encounter-reuseObservationFunction/patients/patient-bundle.json'
+);
+
+describe('ratio Encounter reuse observation function measure', () => {
+  it('does not put all results on observation usage of a function on one observation in population results', async () => {
+    const results = await calculate(MEASURE_BUNDLE, [PATIENT_BUNDLE], CALCULATION_OPTIONS);
+    if (!results.results[0].detailedResults) {
+      fail('Population DetailedResults not found');
+    }
+    const popResults = results.results[0].detailedResults[0].populationResults as PopulationResult[];
+    const numerResults = popResults.find(r => r.populationType === PopulationType.NUMER);
+    const denomResults = popResults.find(r => r.populationType === PopulationType.DENOM);
+    if (!numerResults || !denomResults) {
+      fail('NUMER or DENOM results not found');
+    }
+    expect(numerResults.populationId).toBeDefined();
+    expect(denomResults.populationId).toBeDefined();
+    const numerObservResults = popResults.find(
+      r => r.populationType === PopulationType.OBSERV && r.criteriaReferenceId === numerResults.populationId
+    );
+    const denomObservResults = popResults.find(
+      r => r.populationType === PopulationType.OBSERV && r.criteriaReferenceId === denomResults.populationId
+    );
+    expect(numerObservResults).toBeDefined();
+    expect(denomObservResults).toBeDefined();
+    expect(numerObservResults?.observations).toEqual([1]);
+    expect(denomObservResults?.observations).toEqual([1]);
```

Comment:
Not strictly necessary, but just thinking ahead here. Do you think it would be beneficial to have a test helper that allows for repeatable assertions of observation results? Similar to what we did for `assertPopulationResults`? Then maybe there's a world where we can do something like (pseudocode):

```ts
assertMeasureObservation(groupInfo, criteriaReferenceId, [1])
```

Reply:
Implemented this in 435be92. Made it so the criteria reference is passed in by the population type of the referenced criteria. The helper function will look up that population by type, then use the id to find the right observation to assert on.